CMakeLists.txt

```diff
@@ -0,0 +1,178 @@
+    async_grpc/common/port.h
+    async_grpc/common/mutex.h
+    async_grpc/common/time.h
+    async_grpc/service.h
+    async_grpc/execution_context.h
+    async_grpc/completion_queue_thread.h
+    async_grpc/rpc_handler_interface.h
+    async_grpc/rpc_handler.h
+    async_grpc/type_traits.h
+    async_grpc/event_queue_thread.h
+    async_grpc/retry.h
+    async_grpc/client.h
+    async_grpc/rpc.h
+    async_grpc/server.h
+    async_grpc/testing/rpc_handler_test_server.h
+    async_grpc/testing/rpc_handler_wrapper.h
+    async_grpc/rpc_service_method_traits.h)
+
+set(ALL_LIBRARY_SRCS
+    async_grpc/completion_queue_thread.cc
+    async_grpc/server.cc
+    async_grpc/service.cc
+    async_grpc/common/time.cc
+    async_grpc/retry.cc
+    async_grpc/rpc.cc
+    async_grpc/event_queue_thread.cc)
+
+set(ALL_TESTS
+    async_grpc/server_test.cc
+    async_grpc/type_traits_test.cc)
+
+list(REMOVE_ITEM ALL_LIBRARY_SRCS ${ALL_TESTS})
+
+set(INSTALL_SOURCE_HDRS ${ALL_LIBRARY_HDRS})
+
+file(GLOB_RECURSE ALL_PROTOS "async_grpc*/*.proto")
+set(ALL_PROTOS
+    async_grpc/proto/math_service.proto)
+
+set(ALL_PROTO_SRCS)
+set(ALL_PROTO_HDRS)
+foreach(REL_FIL ${ALL_PROTOS})
+  get_filename_component(DIR ${REL_FIL} DIRECTORY)
+  get_filename_component(FIL_WE ${REL_FIL} NAME_WE)
+
+  list(APPEND ALL_PROTO_SRCS "${PROJECT_BINARY_DIR}/${DIR}/${FIL_WE}.pb.cc")
+  list(APPEND ALL_PROTO_HDRS "${PROJECT_BINARY_DIR}/${DIR}/${FIL_WE}.pb.h")
+
+  add_custom_command(
+    OUTPUT "${PROJECT_BINARY_DIR}/${DIR}/${FIL_WE}.pb.cc"
+           "${PROJECT_BINARY_DIR}/${DIR}/${FIL_WE}.pb.h"
+    COMMAND  ${PROTOBUF_PROTOC_EXECUTABLE}
+    ARGS --cpp_out  ${PROJECT_BINARY_DIR} -I
+      ${PROJECT_SOURCE_DIR} ${REL_FIL}
+    DEPENDS ${REL_FIL}
+    COMMENT "Running C++ protocol buffer compiler on ${REL_FIL}"
+    VERBATIM
+  )
+endforeach()
+set_source_files_properties(${ALL_PROTO_SRCS} ${ALL_PROTO_HDRS} PROPERTIES GENERATED TRUE)
+list(APPEND ALL_LIBRARY_HDRS ${ALL_PROTO_HDRS})
+list(APPEND ALL_LIBRARY_SRCS ${ALL_PROTO_SRCS})
+
+add_library(${PROJECT_NAME} ${ALL_LIBRARY_HDRS} ${ALL_LIBRARY_SRCS})
+
+foreach(REL_FIL ${ALL_TESTS})
+  get_filename_component(DIR ${REL_FIL} DIRECTORY)
+  get_filename_component(FIL_WE ${REL_FIL} NAME_WE)
+  # Replace slashes as required for CMP0037.
+  string(REPLACE "/" "." TEST_TARGET_NAME "${DIR}/${FIL_WE}")
+  google_test("${TEST_TARGET_NAME}" ${REL_FIL})
+  target_link_libraries("${TEST_TARGET_NAME}" PUBLIC grpc++)
+endforeach()
+
+target_link_libraries(${PROJECT_NAME} PUBLIC glog)
+target_link_libraries(${PROJECT_NAME} PUBLIC gflags)
+
+target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
+  ${PROTOBUF_INCLUDE_DIR})
+# TODO(hrapp): This should not explicitly list pthread and use
+# PROTOBUF_LIBRARIES, but that failed on first try.
+target_link_libraries(${PROJECT_NAME} PUBLIC ${PROTOBUF_LIBRARY} pthread)
+target_link_libraries(${PROJECT_NAME} PUBLIC grpc++)
+
+# Add the binary directory first, so that port.h is included after it has
```

Comment:
Is port.h really generated? Drop comment.

Reply:
it's not. fixed